### PR TITLE
Fix bugs in logistic regression

### PR DESCRIPTION
**Pull request by vandenman**
Fixes:

https://github.com/jasp-stats/jasp-test-release/issues/556
https://github.com/jasp-stats/jasp-test-release/issues/557
https://github.com/jasp-stats/jasp-test-release/issues/560

The changes are a bit more drastic than I wanted to, but that's what happens I guess. In the unit tests, the columns of one table got reordered (don't ask me why). The other unit test got smaller since the CI is no longer shown by default.

@FBartos can you verify that the issues are fixed?
@TimKDJ can you take a general look at the code to see if I did anything stupid?

### CODE REVIEW

File: JASP-Engine/JASP/R/regressionlogistic.R [hunk 1]
```diff
@@ -226,8 +219,12 @@ RegressionLogistic <- function(jaspResults, dataset = NULL, options, ...) {
   .reglogisticSetError(res, estimatesTableBootstrap)
   
   estimatesTableBootstrap$addFootnote(gettextf("Bootstrapping based on %i replicates.", options$coeffEstimatesBootstrappingReplicates))
-  estimatesTableBootstrap$addFootnote(gettext("Coefficient estimate is based on the median of the bootstrap distribution."))
-  estimatesTableBootstrap$addFootnote(gettext("Bias corrected accelerated."), symbol = "\u002A")
+  estimatesTableBootstrap$addFootnote(ngettext(1 * options$robustSEOpt,
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/regressionlogistic.R]: I'm not sure `ngettext()` is the best solution for this. I think this creates a special "plural" entry type in the .po file, which would not be appropriate here. Can you verify this @FransMeerhoff?

File: JASP-Engine/JASP/R/regressionlogistic.R [hunk 2]
```diff
@@ -599,60 +596,122 @@ RegressionLogistic <- function(jaspResults, dataset = NULL, options, ...) {
...
-          cilo  = as.numeric(bootstrap.ci$bca[4]),
-          ciup  = as.numeric(bootstrap.ci$bca[5]),
           .isNewGroup = as.logical(j == 1)
         )
 
+        if (options$coeffCI) {
+          result.bootstrap.ci <- try(boot::boot.ci(bootstrap.summary, type = "bca", conf = options$coeffCIInterval, index=j))
+          if(!isTryError(result.bootstrap.ci))
+            bootstrap.ci <- result.bootstrap.ci
+          else if(identical(attr(result.bootstrap.ci, "condition")$message, "estimated adjustment 'a' is NA")){ #Does this still work if we turn on translations? I'm gonna guess not...
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/regressionlogistic.R]: It'll only stop working if the package uses `gettext()` and includes a .po file for the language.. I suppose we made error message checking even more fragile ;)

File: JASP-Engine/JASP/R/regressionlogistic.R [hunk 3]
```diff
@@ -195,28 +184,32 @@ RegressionLogistic <- function(jaspResults, dataset = NULL, options, ...) {
...
     estimatesTableBootstrap$addColumnInfo(name = "model", title = gettext("Model"),          type = "string", combine = TRUE)
   estimatesTableBootstrap$addColumnInfo(name = "param",   title = paramtitle,                type = "string")
   estimatesTableBootstrap$addColumnInfo(name = "est",     title = gettext("Estimate"),       type = "number", format="dp:3")
   estimatesTableBootstrap$addColumnInfo(name = "bias",    title = gettext("Bias"),           type = "number", format="dp:3")
-  estimatesTableBootstrap$addColumnInfo(name = "se",      title = gettext("Standard Error"), type = "number", format="dp:3")
-  estimatesTableBootstrap$addColumnInfo(name = "cilo",    title = gettext("Lower bound"),    type = "number", format="dp:3", overtitle = ciTitle)
-  estimatesTableBootstrap$addColumnInfo(name = "ciup",    title = gettext("Upper bound"),    type = "number", format="dp:3", overtitle = ciTitle)
+  estimatesTableBootstrap$addColumnInfo(name = "se",      title = seTitle,                   type = "number", format="dp:3")
+  if(options$stdCoeff)
+    estimatesTableBootstrap$addColumnInfo(name = "std",   title = gettextf("Standardized%s", "\u207A"), type = "number", format="dp:3")
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/regressionlogistic.R:206]: Good to add the footnote too, if you add the symbol for the footnote (not sure if people would look at different tables for the symbol)

**Review by TimKDJ — COMMENTED** [review]
Some minor suggestions, looks good in general. I couldn't really judge the additional statistics for the bootstrapping though

**Review by FBartos — APPROVED** [review]

File: JASP-Engine/JASP/R/regressionlogistic.R [hunk 4]
```diff
@@ -219,12 +222,10 @@ RegressionLogistic <- function(jaspResults, dataset = NULL, options, ...) {
...
-  estimatesTableBootstrap$addFootnote(ngettext(1 * options$robustSEOpt,
-                                               "Coefficient estimate and robust standard error are based on the median of the bootstrap distribution.",
-                                               "Coefficient estimate is based on the median of the bootstrap distribution."
-                                               ))
-  if (options$coeffCI)
-    estimatesTableBootstrap$addFootnote(gettext("Bias corrected accelerated."), symbol = "\u002A")
+  if (options$robustSEOpt)
+    estimatesTableBootstrap$addFootnote("Coefficient estimate and robust standard error are based on the median of the bootstrap distribution.")
+  else
+    estimatesTableBootstrap$addFootnote("Coefficient estimate is based on the median of the bootstrap distribution.")
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/regressionlogistic.R]: 2x `gettext()`

File: JASP-Engine/JASP/R/regressionlogistic.R [hunk 5]
```diff
@@ -1105,3 +1166,27 @@ RegressionLogistic <- function(jaspResults, dataset = NULL, options, ...) {
   
   return(p)
 }
+
+.reglogisticEstimatesInfo <- function(options, addBCA = FALSE) {
+  # so we only translate this once
+  ciTitle <- sprintf(ngettext(1*addBCA,
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/regressionlogistic.R]: I'd un-`ngettext()` this one too

**Review by TimKDJ — CHANGES_REQUESTED** [review]

**Review by TimKDJ — APPROVED** [review]

**Comment by TimKDJ** [comment]
Thanks for taking care of the logistic regression issues!

**Pull request merged**